src/dialog/dialog.service.ts

```diff
@@ -78,11 +78,9 @@ export class DialogService {
 			dialogRef = viewContainer.createComponent(componentFactory, 0, this.injector);
 		} else if (!this.placeholderService.hasPlaceholderRef()) {
 			dialogRef = viewContainer.createComponent(componentFactory, 0, this.injector);
-			setTimeout(() => {
-				if (dialogRef) {
-					window.document.querySelector("body").appendChild(dialogRef.location.nativeElement);
-				}
-			});
+			if (dialogRef) {
```

Comment:
Is there a reasonable explanation as to how this would stop the dialog from staying open?

Reply:
I'm still not 100% sure why, removing the set timeout prevents dialogs from staying open if I try and toggle them as fast as I can. Increasing the timeout makes it easier to reproduce the problem

Reply:
Huh 🤔 It's possible this was a fix for something that isn't broken anymore

Reply:
Adding a `setTimeout` to the close also seems to prevent it from staying open... I think that's a safer approach to avoid any Expression changed errors which can potentially pop up

Reply:
That sounds like a potentially better solution 👍